azalea-protocol/src/connect.rs

```diff
@@ -323,4 +323,24 @@ where
             },
         }
     }
+
+    pub fn wrap(stream: TcpStream) -> Connection<R1, W1> {
```

Comment:
LGTM, but can you add a doc comment for the function explaining what it does / what it's meant to be used for?